### ISSUE TITLE
stac_search can't find source

### ISSUE THREAD

**Issue by zxdawn** [opengeos/leafmap#562]
### Environment Information

-   leafmap version: 0.26.0
-   Python version: 3.9.15
-   Operating System: Mac

### Description

I tried to use `stac_search` to get different attributes, but it failed. Here's one example of getting gdf.

### What I Did

```
import leafmap
from pystac_client import Client

url = 'https://cmr.earthdata.nasa.gov/stac/LPCLOUD/'
collections = ['EMITL1BRAD_001']
bbox = [-99.65, 18.85, -98.5, 19.95]
date_range = "2022-05/2023-08"


leafmap_search = leafmap.stac_search(
    url=url,
    collections=collections,
    bbox=bbox,
    datetime=date_range,
    get_gdf=True,
)
```

Error:
```
File [~/miniconda3/envs/emit_tutorials/lib/python3.9/site-packages/leafmap/stac.py:1375](https://file+.vscode-resource.vscode-cdn.net/Users/xinz/Documents/github/emit/notebook/~/miniconda3/envs/emit_tutorials/lib/python3.9/site-packages/leafmap/stac.py:1375), in stac_search(url, method, max_items, limit, ids, collections, bbox, intersects, datetime, query, filter, filter_lang, sortby, fields, get_collection, get_items, get_assets, get_links, get_gdf, get_info, **kwargs)
   1371 elif get_gdf:
   1372     import geopandas as gpd
   1374     gdf = gpd.GeoDataFrame.from_features(
-> 1375         search.item_collection().to_dict(), crs="EPSG:4326"
   1376     )
   1377     return gdf
   1378 elif get_info:

File [~/miniconda3/envs/emit_tutorials/lib/python3.9/site-packages/pystac_client/item_search.py:765](https://file+.vscode-resource.vscode-cdn.net/Users/xinz/Documents/github/emit/notebook/~/miniconda3/envs/emit_tutorials/lib/python3.9/site-packages/pystac_client/item_search.py:765), in ItemSearch.item_collection(self)
    757 """
    758 Get the matching items as a :py:class:`pystac.ItemCollection`.
    759 
...
--> 217     raise APIError.from_response(resp)
    218 try:
    219     return resp.content.decode("utf-8")

APIError: {"errors":["Oops! Unable to find the requested resource."]}
```

I also tried to use pystac directly:
```
catalog = Client.open(url)

search = catalog.search(
    collections=collections,
    bbox=bbox,
    datetime=date_range,
    limit=100
)

print('Matching STAC Items:', search.matched())

# search.item_collection().to_dict()
import geopandas as gpd

gdf = gpd.GeoDataFrame.from_features(
                search.item_collection().to_dict(), crs="EPSG:4326"
            )

print(gdf)
```

It works well:
```
Matching STAC Items: 16
                                             geometry              datetime  \
0   POLYGON ((-99.46811 19.87512, -100.04282 19.39...  2023-02-16T21:25:01Z   
1   POLYGON ((-98.98611 19.29643, -99.56340 18.821...  2023-02-16T21:25:13Z   
2   POLYGON ((-99.15800 20.59029, -99.73640 20.101...  2023-02-24T18:14:17Z   
3   POLYGON ((-98.67123 20.01146, -99.25066 19.527...  2023-02-24T18:14:29Z   
4   POLYGON ((-98.18805 19.43193, -98.76950 18.951...  2023-02-24T18:14:41Z   
5   POLYGON ((-98.85368 20.52248, -99.42998 20.035...  2023-02-28T16:39:54Z 
...
```

So, I suppose there's something wrong in leafmap. I checked the parameters used in leafmap's `client`:
```
method POST
max_items None
limit 100
ids None
collections ['EMITL1BRAD_001']
bbox [-99.65, 18.85, -98.5, 19.95]
intersects None
datetime 2022-05/2023-08
query None
filter None
filter_lang None
sortby None
fields None
```

It seems fine. Any ideas?

**Comment by zxdawn** [comment]
OK. Find out the issue: This [line](https://github.com/opengeos/leafmap/blob/1149c9c45ba775dd3409ccc1d317196bcaf9d085/leafmap/stac.py#L1181) gets the root of url (https://cmr.earthdata.nasa.gov/stac) instead of the url input (https://cmr.earthdata.nasa.gov/stac/LPCLOUD).

**Comment by zxdawn** [comment]
Why does leafmap force using the root of url? @giswqs I suppose it's reasonable to use the input url directly.

**Comment by giswqs** [comment]
Your sample code does not work for me. 

```python
from pystac_client import Client

url = 'https://cmr.earthdata.nasa.gov/stac/LPCLOUD/'
collections = ['EMITL1BRAD_001']
bbox = [-99.65, 18.85, -98.5, 19.95]
date_range = "2022-05/2023-08"


catalog = Client.open(url)

search = catalog.search(
    collections=collections,
    bbox=bbox,
    datetime=date_range,
    limit=100
)

print('Matching STAC Items:', search.matched())

# search.item_collection().to_dict()
import geopandas as gpd

gdf = gpd.GeoDataFrame.from_features(
                search.item_collection().to_dict(), crs="EPSG:4326"
            )

print(gdf)
```
![image](https://github.com/opengeos/leafmap/assets/5016453/86a5623b-a8b4-4b48-9c97-e11600cef857)

**Comment by zxdawn** [comment]
The NASA service was down at that time, it should work now.

**Comment by zxdawn** [comment]
Well, the AWS s3 is designed for downloading data.

I have contacted CMR support for the stac issue:
<img width="460" alt="image" src="https://github.com/opengeos/leafmap/assets/30388627/10188b35-3e60-4069-872a-d52ce3ffd8d1">

The earthdata website is also down ....
<img width="1216" alt="image" src="https://github.com/opengeos/leafmap/assets/30388627/50ff9142-67a3-44ad-8c24-c1b4d6391241">

I will test your PR when they fix the link.

**Comment by zxdawn** [comment]
According to their [reply](https://forum.earthdata.nasa.gov/viewtopic.php?t=4683&sid=d230d73bcccdab17059c7a426e16fb21), they have moved the url to page2. So, I have tested the new url:
```
from pystac_client import Client
import geopandas as gpd

url = 'https://cmr.earthdata.nasa.gov/stac/LPCLOUD/?page=2'
collections = ['EMITL1BRAD.v001']
bbox = [-99.65, 18.85, -98.5, 19.95]
date_range = "2022-05/2023-08"

catalog = Client.open(url)

search = catalog.search(
collections=collections,
bbox=bbox,
datetime=date_range,
limit=100
)

print('Matching STAC Items:', search.matched())

gdf = gpd.GeoDataFrame.from_features(
                search.item_collection().to_dict(), crs="EPSG:4326"
            )
```

It works well. Then, I tested your modified leafmap version:
```
import leafmap

leafmap_search = leafmap.stac_search(
    url=url,
    collections=collections,
    bbox=bbox,
    datetime=date_range,
    get_gdf=True,
)
```

I got this error:
```
---------------------------------------------------------------------------
DoesNotConformTo                          Traceback (most recent call last)
[/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb](https://file+.vscode-resource.vscode-cdn.net/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb) Cell 5 line 3
      [1](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=0) import leafmap
----> [3](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=2) leafmap_search = leafmap.stac_search(
      [4](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=3)     url=url,
      [5](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=4)     collections=collections,
      [6](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=5)     bbox=bbox,
      [7](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=6)     datetime=date_range,
      [8](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=7)     get_gdf=True,
      [9](vscode-notebook-cell:/Users/xinz/Documents/github/enmap/notebooks/EnMAP_satpy.ipynb#X21sZmlsZQ%3D%3D?line=8) )

File [~/Documents/github/leafmap_stac/leafmap/stac.py:1346](https://file+.vscode-resource.vscode-cdn.net/Users/xinz/Documents/github/enmap/notebooks/~/Documents/github/leafmap_stac/leafmap/stac.py:1346), in stac_search(url, method, max_items, limit, ids, collections, bbox, intersects, datetime, query, filter, filter_lang, sortby, fields, get_collection, get_items, get_assets, get_links, get_gdf, get_info, get_root, **kwargs)
   1343     collections = [collection_id]
   1345 print(method, max_items, limit, ids, collections, bbox, intersects, datetime, query, filter, filter_lang, sortby, fields)
-> 1346 search = client.search(
   1347     method=method,
   1348     max_items=max_items,
   1349     limit=limit,
   1350     ids=ids,
   1351     collections=collections,
   1352     bbox=bbox,
   1353     intersects=intersects,
   1354     datetime=datetime,
   1355     query=query,
...
   (...)
    612     modifier=self.modifier,
    613 )

DoesNotConformTo: Server does not conform to ITEM_SEARCH, There is not fallback option available for search.
```

**Comment by giswqs** [comment]
Sorry for the long delay. This issue has been fixed in #564

```python
import leafmap

url = 'https://cmr.earthdata.nasa.gov/stac/LPCLOUD/?page=2'
collections = ['EMITL1BRAD.v001']
bbox = [-99.65, 18.85, -98.5, 19.95]
date_range = "2022-05/2023-08"

gdf = leafmap.stac_search(
    url=url,
    collections=collections,
    bbox=bbox,
    datetime=date_range,
    get_gdf=True,
    get_root=False
)
gdf
```
